runtime/src/snapshot_utils.rs

```diff
@@ -1608,25 +1684,26 @@ pub fn verify_snapshot_archive<P, Q, R>(
 }
 
 /// Remove outdated bank snapshots
-pub fn purge_old_bank_snapshots<P>(bank_snapshots_dir: P)
-where
-    P: AsRef<Path>,
-{
-    let mut bank_snapshot_infos = get_bank_snapshots(&bank_snapshots_dir);
-    bank_snapshot_infos.sort_unstable();
-    bank_snapshot_infos
-        .into_iter()
-        .rev()
-        .skip(MAX_BANK_SNAPSHOTS_TO_RETAIN)
-        .for_each(|bank_snapshot_info| {
-            let r = remove_bank_snapshot(bank_snapshot_info.slot, &bank_snapshots_dir);
-            if r.is_err() {
-                warn!(
-                    "Couldn't remove bank snapshot at: {}",
-                    bank_snapshot_info.snapshot_path.display()
-                );
-            }
-        })
+pub fn purge_old_bank_snapshots(bank_snapshots_dir: impl AsRef<Path>) {
+    let do_purge = |mut bank_snapshots: Vec<BankSnapshotInfo>| {
+        bank_snapshots.sort_unstable();
+        bank_snapshots
+            .into_iter()
+            .rev()
+            .skip(MAX_BANK_SNAPSHOTS_TO_RETAIN)
+            .for_each(|bank_snapshot| {
+                let r = remove_bank_snapshot(bank_snapshot.slot, &bank_snapshots_dir);
+                if r.is_err() {
+                    warn!(
+                        "Couldn't remove bank snapshot at: {}",
```

Comment:
nit: maybe also include pre / post information indicating which one is failing.

Reply:
Since the path is displayed, that should be sufficient to indicate which one is failing, yes? `Pre` will have the `.pre` extension.